src/common/blib_math.c

```diff
@@ -232,6 +232,37 @@ var_num_t statmeandev(var_num_t *e, int count) {
   return sum / count;
 }
 
+//
+// compare function for median calculation
+//
+int median_compare(const void * a, const void * b) {
+  var_num_t fa = *(var_num_t*) a;
+  var_num_t fb = *(var_num_t*) b;
+  return (fa > fb) - (fa < fb);
+}
+
+//
+// Median
+//
+// Median is calculated using quicksort with a complexity of O(n log n).
+// Faster approch would be to implement i.e. quickselect with a 
+// complexity of O(n).
+//
+var_num_t statmedian(var_num_t *e, int count) {  
+  if (count == 0) {
+    return 0;
+  }
+
+  qsort(e, count, sizeof(var_num_t), median_compare);
+
+  if (count % 2 == 0) {
+    return (e[count/2] + e[count/2 - 1]) / 2;
+  }
```

Comment:
sorry to be a pain, but could you pls reformat the `else` line to be on the same line as the closing brace, ie:

`} else {`

Reply:
Cool stuff  - thanks for the changes